.github/workflows/ci.yaml

```diff
@@ -0,0 +1,108 @@
+name: CI Workflow
+
+on: [push, pull_request, workflow_dispatch]
+
+# Separate build and deploy jobs so we can build all branches and pull requests
+# but only deploy from main
+jobs:
+  build:
+    name: Build & Test
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+
+      - uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+          ruby-version: '2.7'
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Build
+        run: bundle exec middleman build
+
+      - name: Test
+        run: bundle exec rake
+
+      - name: Check links
+        run: npm run check-links
+
+  # Deploy Frontend Docs to production when the main branch is changed
+  # Github Actions is not involved in deploying PR or branch previews – these are handled by Netlify
+  # Only runs if the production environment protection rules are fulfilled
+  deploy:
+    environment: production
```

Comment:
@lfdebrux @hannalaakso What is this being used for?

Reply:
The environment provides an extra level of security for secrets, see https://github.com/alphagov/design-system-team-internal/issues/446. It also is the thing that causes GitHub to create a deployment notification on Slack.